tests/device_tests/test_msg_signtx_external.py

```diff
@@ -0,0 +1,559 @@
+# This file is part of the Trezor project.
+#
+# Copyright (C) 2020 SatoshiLabs and contributors
+#
+# This library is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License version 3
+# as published by the Free Software Foundation.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the License along with this library.
+# If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.
+
+import pytest
+
+from trezorlib import btc, messages as proto
+from trezorlib.exceptions import TrezorFailure
+from trezorlib.tools import parse_path
+
+from ..tx_cache import TxCache
+from .signtx import request_finished, request_input, request_meta, request_output
+
+B = proto.ButtonRequestType
+
+TX_CACHE_TESTNET = TxCache("Testnet")
+TX_CACHE_MAINNET = TxCache("Bitcoin")
+
+TXHASH_e5040e = bytes.fromhex(
+    "e5040e1bc1ae7667ffb9e5248e90b2fb93cd9150234151ce90e14ab2f5933bcd"
+)
+TXHASH_d830b8 = bytes.fromhex(
+    "d830b877c3d9237a0a68be88825a296da01ac282a2efd2f671d8f17f15117b74"
+)
+TXHASH_091446 = bytes.fromhex(
+    "09144602765ce3dd8f4329445b20e3684e948709c5cdcaf12da3bb079c99448a"
+)
+TXHASH_65b811 = bytes.fromhex(
+    "65b811d3eca0fe6915d9f2d77c86c5a7f19bf66b1b1253c2c51cb4ae5f0c017b"
+)
+TXHASH_e5b7e2 = bytes.fromhex(
+    "e5b7e21b5ba720e81efd6bfa9f854ababdcddc75a43bfa60bf0fe069cfd1bb8a"
+)
+TXHASH_70f987 = bytes.fromhex(
+    "70f9871eb03a38405cfd7a01e0e1448678132d815e2c9f552ad83ae23969509e"
+)
+TXHASH_65b768 = bytes.fromhex(
+    "65b768dacccfb209eebd95a1fb80a04f1dd6a3abc6d7b41d5e9d9f91605b37d9"
+)
+TXHASH_a345b8 = bytes.fromhex(
+    "a345b85759b385c6446055e4c3baa77e8161a65009dc009489b48aa6587ce348"
+)
+
+
+@pytest.mark.skip_t1
+class TestMsgSigntxExternal:
```

Comment:
please remove the class and move tests to top level

Reply:
4109bf561d53ecd2991a971e270f1a2b6ee6a674